deployments/spacelift/dpe-k8s/main.tf

```diff
@@ -0,0 +1,188 @@
+locals {
+  k8s_stack_environment_variables = {
+    aws_account_id                    = var.aws_account_id
+    region                            = var.region
+    pod_security_group_enforcing_mode = var.pod_security_group_enforcing_mode
+    cluster_name                      = var.cluster_name
+    vpc_name                          = var.vpc_name
+    vpc_cidr_block                    = var.vpc_cidr_block
+    public_subnet_cidrs               = var.public_subnet_cidrs
+    private_subnet_cidrs              = var.private_subnet_cidrs
+    azs                               = var.azs
+  }
+
+  k8s_stack_deployments_variables = {
+    spotinst_account = var.spotinst_account
+    vpc_cidr_block   = var.vpc_cidr_block
+  }
+
+  # Variables to be passed from the k8s stack to the deployments stack
+  k8s_stack_to_deployment_variables = {
+    vpc_id                 = "TF_VAR_vpc_id"
+    private_subnet_ids     = "TF_VAR_private_subnet_ids"
+    node_security_group_id = "TF_VAR_node_security_group_id"
+    pod_to_node_dns_sg_id  = "TF_VAR_pod_to_node_dns_sg_id"
+  }
+}
+
+resource "spacelift_space" "dpe-space" {
+  name             = var.space_name
+  parent_space_id  = var.parent_space_id
+  description      = "Contains resources for the DPE team."
+  inherit_entities = true
+}
+
+resource "spacelift_stack" "k8s-stack" {
+  github_enterprise {
+    namespace = "Sage-Bionetworks-Workflows"
+    id        = "sage-bionetworks-workflows-gh"
+  }
+
+  depends_on = [
+    spacelift_space.dpe-space
+  ]
+
+  administrative          = false
+  autodeploy              = var.auto_deploy
+  branch                  = var.git_branch
+  description             = "Infrastructure to support deploying to an EKS cluster"
+  name                    = var.k8s_stack_name
+  project_root            = var.k8s_stack_project_root
+  repository              = "eks-stack"
+  terraform_version       = var.opentofu_version
+  terraform_workflow_tool = "OPEN_TOFU"
+  space_id                = spacelift_space.dpe-space.id
+}
+
+resource "spacelift_environment_variable" "k8s-stack-environment-variables" {
+  for_each = local.k8s_stack_environment_variables
+
+  stack_id   = spacelift_stack.k8s-stack.id
+  name       = "TF_VAR_${each.key}"
+  value      = try(tostring(each.value), jsonencode(each.value))
```

Comment:
This logic was a bit unfortunate to have to implement but it's to get around the fact that complex types being stored as environment variable as a bit wonky. See this reading:

1. https://docs.spacelift.io/vendors/terraform/storing-complex-variables
2. https://developer.hashicorp.com/terraform/language/functions/try

Without having the try and attempting to convert it to a string first I got this error:
`Invalid AWS Region: "us-east-1"`

This is because the environment variable was stored as:
`"us-east-1"`
instead of
`us-east-1`